pkg/deploy/api/types.go

```diff
@@ -118,6 +123,14 @@ type ExecNewPodHook struct {
 	Volumes []string
 }
 
+// TagImageHook is a request to tag the image in a particular container onto an ImageStreamTag.
+type TagImageHook struct {
+	// ContainerName is the name of a container in the deployment config whose image value will be used as the source of the tag
+	ContainerName string
```

Comment:
> Is defaultable, don't see it being necessary to default it.  I'd rather
> people be explicit.  However, I will validate it for people

The entire API and CLI are based around the idea of specifying as little information as possible.  Why is this case special?  The "normal" flow is a DC with a single container.